airbyte/cloud/_workspaces.py

```diff
@@ -0,0 +1,351 @@
+        """Deploy a source and cache to the workspace as a new connection.
+
+        Returns the newly deployed connection ID as a `str`.
+
+        Args:
+            source (Source | str): The source to deploy. You can pass either an already deployed
+                source ID `str` or a PyAirbyte `Source` object. If you pass a `Source` object,
+                it will be deployed automatically.
+            cache (CacheBase, optional): The cache to deploy as a new destination. You can provide
+                `cache` or `destination`, but not both.
+            destination (str, optional): The destination ID to use. You can provide
+                `cache` or `destination`, but not both.
+        """
+        # Resolve source ID
+        source_id: str
+        if isinstance(source, Source):
+            selected_streams = selected_streams or source.get_selected_streams()
+            if source._deployed_source_id:  # noqa: SLF001
+                source_id = source._deployed_source_id  # noqa: SLF001
+            else:
+                source_id = self.deploy_source(source)
+        else:
+            source_id = source
+            if not selected_streams:
+                raise exc.PyAirbyteInputError(
+                    guidance="You must provide `selected_streams` when deploying a source ID."
+                )
+
+        # Resolve destination ID
+        destination_id: str
+        if destination:
+            destination_id = destination
+        elif cache:
+            table_prefix = table_prefix if table_prefix is not None else (cache.table_prefix or "")
+            if not cache._deployed_destination_id:  # noqa: SLF001
+                destination_id = self.deploy_cache_as_destination(cache)
+            else:
+                destination_id = cache._deployed_destination_id  # noqa: SLF001
+        else:
+            raise exc.PyAirbyteInputError(
+                guidance="You must provide either a destination ID or a cache object."
+            )
+
+        assert source_id is not None
+        assert destination_id is not None
+
+        deployed_connection = create_connection(
+            name="Connection (Deployed by PyAirbyte)",
+            source_id=source_id,
+            destination_id=destination_id,
+            api_root=self.api_root,
+            api_key=self.api_key,
+            workspace_id=self.workspace_id,
+            selected_stream_names=selected_streams,
+            prefix=table_prefix or "",
+        )
+
+        if isinstance(source, Source):
+            source._deployed_connection_id = deployed_connection.connection_id  # noqa: SLF001
+        if cache:
+            cache._deployed_connection_id = deployed_connection.connection_id  # noqa: SLF001
+
+        return deployed_connection.connection_id
+
+    def delete_connection(
+        self,
+        connection_id: str | None,
+        *,
+        delete_source: bool = False,
+        delete_destination: bool = False,
+    ) -> None:
```

Comment:
Might be worth adding a deletion flag to this class's private attributes. Then, other methods which should not operate on a deleted object could fail in a sensible way. (Might be cleanest to implement this with a decorator on those methods.)